src/App.tsx

```diff
@@ -1,26 +1,51 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { useEffect } from "react"
+
+import { Routes, Route, useNavigate, useLocation } from "react-router-dom"
+import { onAuthStateChanged, getAuth } from "firebase/auth"
+
+import SignupPage from "./pages/signup/signup.component"
+import LoginPage from "./pages/login/login.component"
+import HomePage from "./pages/home/home.component"
+import Sidebar from "./components/sidebar/sidebar.component"
+import MyBlogs from "./pages/myBlogs/myBlogs"
+import BlogView from "./pages/blogView/blogView.component"
+
+//styles
+import "./App.css"
 
 function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+	const navigate = useNavigate()
+	const location = useLocation()
+
+	useEffect(
+		() => {
+			const auth = getAuth()
+			onAuthStateChanged(auth, (user) => {
+				if (!user)
+					navigate('/login')
+				else {
+					if (location.pathname === '/login' || location.pathname === '/signup')
+						navigate('/')
+				}
+			})
+		},
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[]
+	)
+
+	return (
+		<div className="App">
+			<Routes>
+				<Route path='/' element={<Sidebar />}>
+					<Route index={true} element={<HomePage />} />
+					<Route path="my-blogs" element={<MyBlogs />} />
+				</Route>
+				<Route path="/login" element={<LoginPage />}></Route>
```

Comment:
When there is nothing to add in the tag, it is better to make it self closing.
```suggestion
				<Route path="/login" element={<LoginPage />} />
```

Reply:
My bad, sorry. 

Fixed.